packages/react-native/android/src/main/java/backtrace/library/BacktraceFileSystemProvider.java

```diff
@@ -137,16 +134,26 @@ public void exists(String path, Promise promise) {
     }
 
     @ReactMethod(isBlockingSynchronousMethod = true)
-    public boolean renameSync(String sourcePath, String destinationPath) {
-        File sourceFile = new File(sourcePath);
-        File destinationFile = new File(destinationPath);
-        return sourceFile.renameTo(destinationFile);
+    public boolean copySync(String sourcePath, String destinationPath) {
+        return copy(sourcePath, destinationPath);
     }
 
     @ReactMethod
-    public void rename(String sourcePath, String destinationPath, Promise promise) {
-        File sourceFile = new File(sourcePath);
-        File destinationFile = new File(destinationPath);
-        promise.resolve(sourceFile.renameTo(destinationFile));
+    public void copy(String sourcePath, String destinationPath, Promise promise) {
+        promise.resolve(copy(sourcePath, destinationPath));
+    }
+
+    private boolean copy(String sourcePath, String destinationPath) {
+        try (FileInputStream inputStream = new FileInputStream(sourcePath);
+             FileOutputStream outputStream = new FileOutputStream(destinationPath, false)) {
+            byte[] buf = new byte[1024];
+            int len;
+            while ((len = inputStream.read(buf)) > 0) {
+                outputStream.write(buf, 0, len);
+            }
```

Comment:
What Java version do we use here?
You could use `transferTo`: https://docs.oracle.com/javase/9/docs/api/java/io/InputStream.html#transferTo-java.io.OutputStream-

Reply:
![image](https://github.com/backtrace-labs/backtrace-javascript/assets/9386268/ae8c8c1a-948e-42e1-8319-8e9d8b95609d)